notebooks/circuits_with_bloqade.py

```diff
@@ -0,0 +1,763 @@
+
+# %%
+# Option C: write it all in bloqade directly
+try:
+    del bloqade_trotter
+except:
+    pass
+try:
+    del op_zz
+except:
+    pass
+# Define an operator that looks like the ZZ power gate
+@squin.kernel
+def op_zz(theta: float, qb1:bloqade.types.Qubit, qb2:bloqade.types.Qubit):
+    """
+    A kernel that returns an operator that looks like ZZ^{theta/2pi}
+    """
+    squin.gate.cx(qb1, qb2)
+    squin.gate.rz(theta, qb2)
+    squin.gate.cx(qb1, qb2)
+    
+    
+@squin.kernel
+def bloqade_trotter(N:int,
+                    steps:int,
+                    dt:float = 0.01,
+                    J:float = 1,
+                    h:float = 1):
+    """
+    Main function that runs the Trotter circuit for a given number of steps
+    """
+    qubits = squin.qubit.new(N)
+    xpow = squin.op.rot(squin.op.x(), angle=dt * h)
+    for _ in range(steps):
+        for i in range(0, len(qubits)-1):
+            op_zz(theta = dt*J, qb1 = qubits[i], qb2 = qubits[i + 1])
+        for i in range(0, len(qubits)):
+            squin.qubit.apply(operator=xpow, qubits=[qubits[i]])
+    return qubits
+
+
+# %% [markdown]
+# Of course, both Cirq and the (converted) Bloqade kernel have the same execution and same output state.
+
+# %%
+cirq_trotter = trotter_circuit(
+    N=12,
+    steps=10,
+    dt=0.01,
+    J=1,
+    h=1)
+
+cirq_statevector = cirq.Simulator().simulate(cirq_trotter).state_vector()
+
+# Or converting to a bloqade kernel and simulating with PyQrack
+cirq_trotter_kernel = squin.cirq.load_circuit(
+    cirq_trotter,
+    kernel_name="cirq_trotter",
+    register_as_argument=False,
+    return_register=True)
+cirq_trotter_qubits = StackMemorySimulator(min_qubits=12).task(cirq_trotter_kernel).run()
+cirq_bloqade_state = StackMemorySimulator.quantum_state(cirq_trotter_qubits)
+# The state is, of course, the same. A little bit of work is needed to extract out the (single) state vector from the RDM.
+print("Overlap:", np.abs(np.dot(np.conj(cirq_statevector), cirq_bloqade_state.eigenvectors[:,0]))**2)
+
+
+# %% [markdown]
+# Similarly, the execution and output state of the cirq state and kernel written fully in bloqade have the same state. Note that for the `bloqade_trotter` kernel, the arguments are not declared until the simulator is run.
+
+# %%
+# Option C: Bloqade only
+cirq_trotter_qubits = StackMemorySimulator(min_qubits=12).run(bloqade_trotter, args=(12,10,0.01,1,1))
+cirq_bloqade_state = StackMemorySimulator.quantum_state(cirq_trotter_qubits)
+print("Overlap:", np.abs(np.dot(np.conj(cirq_statevector), cirq_bloqade_state.eigenvectors[:,0]))**2)
+
+# %% [markdown]
+# # Mid-circuit feed forward
+# 
+# The powerful thing about bloqade kernels is the ability to natively represent mid-circuit feed-forward using control flow. While the possibilities are endless, including measurement based quantum computing and error correction, we show two examples here.
```

Comment:
You started the previous section with the exact same phrase, maybe "Another cool thing about bloqade kernels..."?